### PR TITLE
Make dumping of reservation info congruent between scrub and recovery

### PR DESCRIPTION
**Pull request by dzafman**
## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/osd/PG.h [hunk 1]
```diff
@@ -1091,7 +1091,7 @@ class PG : public DoutPrefixProvider, public PeeringState::PeeringListener {
 
     // metadata
     set<pg_shard_t> reserved_peers;
-    bool reserved, reserve_failed;
+    bool local_reserved, remote_reserved, reserve_failed;
```
Comment (tchaikov) [inline src/osd/PG.h]: nit, could initialize there member variable in-class.

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -233,8 +233,8 @@ OSDService::OSDService(OSD *osd) :
   publish_lock{ceph::make_mutex("OSDService::publish_lock")},
   pre_publish_lock{ceph::make_mutex("OSDService::pre_publish_lock")},
   max_oldest_map(0),
-  scrubs_pending(0),
-  scrubs_active(0),
+  scrubs_local(0),
```
Comment (tchaikov) [inline src/osd/OSD.cc]: nit, could initialize these member variables in-class.

File: doc/dev/osd_internals/scrub.rst [hunk 3]
```diff
@@ -28,3 +31,11 @@ State variables
 - Initiated scrub state is  must_scrub && !must_deep_scrub && !time_for_deep
 - Initiated scrub after osd_deep_scrub_interval state is must scrub && !must_deep_scrub && time_for_deep
 - Initiated deep scrub state is  must_scrub && must_deep_scrub
+
+Scrub Reservations
+------------------
+
+An OSD daemon command dumps total local and remote reserations::
```
Comment (neha-ojha) [inline doc/dev/osd_internals/scrub.rst]:
```suggestion
An OSD daemon command dumps total local and remote reservations::
```

File: doc/dev/osd_internals/scrub.rst [hunk 4]
```diff
@@ -28,3 +31,11 @@ State variables
...
+
+  ceph daemon osd.<id> dump_scrub_reservations
+
```
Comment (neha-ojha) [inline doc/dev/osd_internals/scrub.rst]: might want to add something similar for `ceph daemon osd.<id> dump_recovery_reservations` in dev/osd_internals/recovery_reservation.rst

File: PendingReleaseNotes [hunk 5]
```diff
@@ -124,3 +124,9 @@
   value determined by the config options, for the average for any of the 3 intervals.
   New admin command ``ceph daemon osd.# dump_osd_network [threshold]`` will
   do the same but only including heartbeats initiated by the specified OSD.
+
+* New OSD daemon command dump_recovery_reservations which reveals the
+  recovery locks held (in_progress) and waiting priority queues.
```
Comment (neha-ojha) [inline PendingReleaseNotes]:
```suggestion
  recovery locks held (in_progress) and waiting in priority queues.
```

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-07_14:44:52-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-08_13:41:17-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-smithi/

**Comment by dzafman** [comment]
Testing of new qa/standalone/scrub/osd-scrub-dump.sh PASSED:

http://pulpito.ceph.com/dzafman-2019-09-11_09:48:02-rados-wip-zafman-testing3-distro-basic-smithi/4298352/

**Pull request merged**